StortSpelProjekt/Engine/src/Renderer/QuadManager.cpp

```diff
@@ -0,0 +1,166 @@
+	vertex.uv = DirectX::XMFLOAT2{ 0.0, 0.0 };
+	vertex.normal = normal;
+	vertex.tangent = tangent;
+	m_Vertices.push_back(vertex);
+
+	vertex.pos = DirectX::XMFLOAT3{ x, size.y, 0.0 };
+	m_Positions["lower_left"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 0.0, 1.0 };
+	m_Vertices.push_back(vertex);
+
+	vertex.pos = DirectX::XMFLOAT3{ size.x, y, 0.0 };
+	m_Positions["upper_right"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 1.0, 0.0 };
+	m_Vertices.push_back(vertex);
+
+	vertex.pos = DirectX::XMFLOAT3{ size.x, size.y, 0.0 };
+	m_Positions["lower_right"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 1.0, 1.0 };
+	m_Vertices.push_back(vertex);
+
+	std::vector<unsigned int> indices = { 0, 1, 2, 1, 2, 3 };
+
+	Renderer* renderer = &Renderer::GetInstance();
+
+	m_pQuad = new Mesh(renderer->m_pDevice5, &m_Vertices, &indices, renderer->m_DescriptorHeaps[DESCRIPTOR_HEAP_TYPE::CBV_UAV_SRV]);
+
+	m_pQuad->m_pUploadResourceVertices = new Resource(renderer->m_pDevice5, m_pQuad->GetSizeOfVertices(), RESOURCE_TYPE::UPLOAD, L"Vertex_UPLOAD_RESOURCE");
+	m_pQuad->m_pDefaultResourceVertices = new Resource(renderer->m_pDevice5, m_pQuad->GetSizeOfVertices(), RESOURCE_TYPE::DEFAULT, L"Vertex_DEFAULT_RESOURCE");
+
+	// Vertices
+	const void* data = static_cast<const void*>(m_pQuad->m_Vertices.data());
+	Resource* uploadR = m_pQuad->m_pUploadResourceVertices;
+	Resource* defaultR = m_pQuad->m_pDefaultResourceVertices;
+
+	// Create SRV
+	D3D12_SHADER_RESOURCE_VIEW_DESC dsrv = {};
+	dsrv.ViewDimension = D3D12_SRV_DIMENSION_BUFFER;
+	dsrv.Buffer.FirstElement = 0;
+	dsrv.Format = DXGI_FORMAT_UNKNOWN;
+	dsrv.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
+	dsrv.Buffer.NumElements = m_pQuad->GetNumVertices();
+	dsrv.Buffer.StructureByteStride = sizeof(Vertex);
+
+	m_pQuad->m_pSRV = new ShaderResourceView(
+		renderer->m_pDevice5,
+		renderer->m_DescriptorHeaps[DESCRIPTOR_HEAP_TYPE::CBV_UAV_SRV],
+		&dsrv,
+		m_pQuad->m_pDefaultResourceVertices);
+
+	// Set indices resource
+	m_pQuad->m_pUploadResourceIndices = new Resource(renderer->m_pDevice5, m_pQuad->GetSizeOfIndices(), RESOURCE_TYPE::UPLOAD, L"Index_UPLOAD_RESOURCE");
+	m_pQuad->m_pDefaultResourceIndices = new Resource(renderer->m_pDevice5, m_pQuad->GetSizeOfIndices(), RESOURCE_TYPE::DEFAULT, L"Index_DEFAULT_RESOURCE");
+
+	// inidices
+	data = static_cast<const void*>(m_pQuad->m_Indices.data());
+	uploadR = m_pQuad->m_pUploadResourceIndices;
+	defaultR = m_pQuad->m_pDefaultResourceIndices;
+
+	// Set indexBufferView
+	m_pQuad->m_pIndexBufferView = new D3D12_INDEX_BUFFER_VIEW();
+	m_pQuad->m_pIndexBufferView->BufferLocation = m_pQuad->m_pDefaultResourceIndices->GetGPUVirtualAdress();
+	m_pQuad->m_pIndexBufferView->Format = DXGI_FORMAT_R32_UINT;
+	m_pQuad->m_pIndexBufferView->SizeInBytes = m_pQuad->GetSizeOfIndices();
+
+	if (m_Clickable)
+	{
+		EventBus::GetInstance().Subscribe(this, &QuadManager::pressed);
+	}
```

Comment:
This should be in Mesh.cpp. I recently wrote this code aswell in setscenedynamic branch so I can fix this later with m_pQuad.init();